addOns/authhelper/src/main/java/org/zaproxy/addon/authhelper/BrowserBasedAuthenticationMethodType.java

```diff
@@ -393,7 +323,7 @@ public WebSession authenticate(
                     }
                 }
 
-                if (authMsg != null) {
+                if (((ClientSideHandler) handler).getAuthMsg() != null) {
```

Comment:
Should keep a local variable for the `authMsg` instead of calling the getter each time.